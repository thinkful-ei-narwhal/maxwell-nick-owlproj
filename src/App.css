.App{ /*Renders main window*/
  height: auto;
  width: 100%;
  text-align: center;
  border: 5px solid;
  border-color: #0000ff;

  display: grid;
  grid-template-columns: 15% 15% 70%;
  grid-template-rows: 100%;
}

