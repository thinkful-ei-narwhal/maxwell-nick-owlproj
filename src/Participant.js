import React from 'react';

export default function Participant(props) {
  console.log(props)

  return (
    <div className='participant' id={props.id}>
      <header>
        <h1>{props.name}</h1>
      </header>
      <img src={props.avatar} alt="testing" />
      <ul >
        <li className={inSession(props.inSession)}>{onStage(props.onStage)}</li>
      </ul>
    </div>
  )
}

function onStage(onStage) {
  if (onStage) {
    return <p>on stage</p>
  }
  else {
    return <p>observer</p>
  }
}

function inSession(inSession) {
  if (inSession) {
    return 'inSessionClass';
  }
  else {
    return 'notInSessionClass'
  }
}