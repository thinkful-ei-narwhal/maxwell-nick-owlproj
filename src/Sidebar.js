import React from 'react';
import Participant from './participant';

import './sidebar.css'

export default function Sidebar(props) {
  const participantComponents = props.participants.map(participant => {
    return <Participant key={participant.id} name={participant.name} avatar={participant.avatar} inSession={participant.inSession} onStage={participant.onStage} />;
  });

  let participantComponentsSorted = [];
  participantComponents.forEach(component => {
    if (component.props.inSession === true) {
      participantComponentsSorted.unshift(component);
    }
    else {
      participantComponentsSorted.push(component);
    }
  });

  return (
    <div className='sidebar'>
      {participantComponentsSorted}
    </div>
  )
}
