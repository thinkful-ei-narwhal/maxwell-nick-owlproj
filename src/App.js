import React from 'react';
import store from './store';
import Sidebar from './Sidebar';
import Stage from './Stage';
import './App.css';

function App() {
  return (
    <main className='App'>
      <Sidebar participants={store.participants} />
      <Stage participants={store.participants} />
    </main>
  );
}

export default App;