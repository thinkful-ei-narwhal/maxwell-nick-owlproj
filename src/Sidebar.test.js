import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import errorRenderer from 'react-test-renderer';

import App from './App';
import store from './store';
import Sidebar from './sidebar';
import Participant from './participant';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

/* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST */
it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
    // render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Sidebar participants={store.participants} />, div);
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });
/* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST *//* SMOKE TEST */
it('renders the UI as expected', () => {
    const newJsonString = errorRenderer.create(<Sidebar participants={store.participants}/>).toJSON();
    expect(newJsonString).toMatchSnapshot();
});
