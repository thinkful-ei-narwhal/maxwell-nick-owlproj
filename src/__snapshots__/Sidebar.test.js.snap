// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="sidebar"
>
  <div
    className="participant"
  >
    <header>
      <h1>
        Ashla Attwool
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <ul>
      <li
        className="inSessionClass"
      >
        <p>
          on stage
        </p>
      </li>
    </ul>
  </div>
  <div
    className="participant"
  >
    <header>
      <h1>
        Frank Runciman
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <ul>
      <li
        className="inSessionClass"
      >
        <p>
          observer
        </p>
      </li>
    </ul>
  </div>
  <div
    className="participant"
  >
    <header>
      <h1>
        Caty Flucker
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <ul>
      <li
        className="inSessionClass"
      >
        <p>
          observer
        </p>
      </li>
    </ul>
  </div>
  <div
    className="participant"
  >
    <header>
      <h1>
        Koren Templeton
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <ul>
      <li
        className="inSessionClass"
      >
        <p>
          on stage
        </p>
      </li>
    </ul>
  </div>
  <div
    className="participant"
  >
    <header>
      <h1>
        Axe Kubicka
      </h1>
    </header>
    <img
      alt="testing"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <ul>
      <li
        className="notInSessionClass"
      >
        <p>
          observer
        </p>
      </li>
    </ul>
  </div>
</div>
`;
